src/components/Sublist3r/Sublist3r.tsx

```diff
@@ -0,0 +1,193 @@
+
+    // Component Constants
+    const title = "Sublist3r";
+    const description = "Sublist3r is a subdomain enumeration tool that gathers subdomains for a given domain.";
+    const steps =
+        "Step 1: Provide the target domain to enumerate subdomains.\n" +
+        "Step 2: Start the enumeration process to gather subdomains.\n" +
+        "Step 3: Review the enumerated subdomains for further analysis.\n";
+    const sourceLink = "https://github.com/aboul3la/Sublist3r"; // Link to the source code
+    const tutorial = "https://github.com/aboul3la/Sublist3r/wiki"; // Link to the official documentation/tutorial
+    const dependencies = ["sublist3r"]; // Contains the dependencies required by the component.
+
+    // Form hook to handle form input
+    let form = useForm({
+        initialValues: {
+            targetDomain: "",
+        },
+    });
+
+    /**
+     * handleProcessData: Callback to handle and append new data from the child process to the output.
+     * It updates the state by appending the new data received to the existing output.
+     * @param {string} data - The data received from the child process.
+     */
+    const handleProcessData = useCallback((data: string) => {
+        setOutput((prevOutput) => prevOutput + "\n" + data); // Append new data to the previous output.
+    }, []);
+
+    /**
+     * handleProcessTermination: Callback to handle the termination of the child process.
+     * Once the process termination is handled, it clears the process PID reference and
+     * deactivates the loading overlay.
+     * @param {object} param - An object containing information about the process termination.
+     * @param {number} param.code - The exit code of the terminated process.
+     * @param {number} param.signal - The signal code indicating how the process was terminated.
+     */
+    const handleProcessTermination = useCallback(
+        ({ code, signal }: { code: number; signal: number }) => {
+            // If the process was successful, display a success message.
+            if (code === 0) {
+                handleProcessData("\nProcess completed successfully.");
+
+                // If the process was terminated manually, display a termination message.
+            } else if (signal === 15) {
+                handleProcessData("\nProcess was manually terminated.");
+
+                // If the process was terminated with an error, display the exit and signal codes.
+            } else {
+                handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+
+            // Clear the child process pid reference. There is no longer a valid process running.
+            setPid("");
+
+            // Cancel the loading overlay. The process has completed.
+            setLoading(false);
+        },
+        [handleProcessData] // Dependency on the handleProcessData callback
+    );
+
+    useEffect(() => {
```

Comment:
This needs to be place above handleProcessTermination and handleProcessData. I am not sure if it affects the app, but it will keep it consistent with the other tools.